### DISCUSSION TITLE
Looking for a simple and clear way to filter for DNS queries in SO 2.3.2

### DISCUSSION THREAD

**Discussion by McSh4dow** [Security-Onion-Solutions/securityonion#1994] [Unsupported Versions] [answered]
Hello there,

I recently upgraded my Security Onion 16.0.4 to 2.3.2 with Suricata and it is working fine so far.

Just one thing...
If a detection is triggered I would like to see the last few DNS queries. With 16.0.4 there was a simple way to do so with Bro.

Did you guys find a way to solve this 'problem'? Do I need to reinstall with Zeek or is there something different I could try?

Thank you in advance!

**Comment by dougburks** [comment, marked as answer]
Are you saying that for metadata you chose Suricata instead of Zeek?  If so, Suricata should be recording DNS queries. Try pivoting to Hunt or Kibana and looking for DNS queries there:
https://docs.securityonion.net/en/2.3/hunt.html
https://docs.securityonion.net/en/2.3/kibana.html

**Reply by McSh4dow** [reply]
Yeah Suricata worked fine with Bro in combination in 16.0.4. But now I was only able to choose Suricata or Zeek. In 2.3.2 Suricata is recording DNS queries as well but I didn't find a way to filter for specific ones or for DNS as protocol. It seems like there is no tag for the DNS queries in Hunt in combination with Suricata.

**Reply by dougburks** [reply]
A couple of things to clarify:
- Bro has been renamed to Zeek. Both Security Onion 16.04 and Security Onion 2.3.2 are using Zeek.
- In Security Onion 16.04, Suricata was only used for alerts and not metadata. In Security Onion 2.3.2, Suricata is used for alerts but can optionally be used for metadata as well (instead of Zeek).
- The Hunt queries and Kibana dashboards are designed to support metadata from either Suricata or Zeek.

I've verified that choosing Suricata for metadata shows DNS queries in Hunt correctly for me:
![image](https://user-images.githubusercontent.com/1659467/99704841-eb70c880-2a66-11eb-8d57-7846d061e85c.png)

When you choose a DNS query in Hunt, are there no results at all?

Have you checked the DNS dashboard in Kibana?

**Reply by McSh4dow** [reply]
Okay that's helping already. But in this case you filtered for dns queries itselves. 
Is it also possible to set a filter on a srcip to show the last few dns queries of this ip when a detection is triggered?

When I search for dns queries and add the source ip to the filter I do not see a clear list of the queries - only the complete metadata.

I might have to check the DNS dashboard in Kibana again.

**Reply by dougburks** [reply]
> Is it also possible to set a filter on a srcip to show the last few dns queries of this ip when a detection is triggered?

Yes, I typically start from an alert, click the IP address in question to bring up the quick action bar, click the Hunt icon to pivot to Hunt and see all logs for that IP grouped by type, then click DNS to filter to just DNS logs for that IP. We cover pivoting like this in the Security Onion Essentials video series:
https://securityonion.net/training

> When I search for dns queries and add the source ip to the filter I do not see a clear list of the queries - only the complete metadata.

I think what you may be saying is that you're seeing the full logs but not the Group Metrics section as shown in my screenshot above. If that's the case, try adding the `groupby` options as shown in my screenshot. You can also group by any arbitrary field by drilling into a log and then clicking the groupby icon as described at https://docs.securityonion.net/en/2.3/hunt.html#events.

**Reply by McSh4dow** [reply]
Okay I will try both tomorrow!
Thank you very much for your help!

**Reply by McSh4dow** [reply]
Hey,
I've tested it today and decided to reinstall with Zeek for metadata.
Currently it's working fine but I got one more question.

Is it still possible to choose a policy for TALOS ruleset somehow? If it is - where can I do that?
Right now it seems like I am logging every single match of the TALOS all.rules and it's flooding my fast.log of Suricata.

I'd like to log the detections that match with the policy "balanced-ips". I'm about to write a script to check a detection's sid and compare it with all.rules but a valid solution would be nice.

In a documentation of 14.04 it was possible to choose a policy right after selection the ruleset (e.g. connectivity, balanced, security).

Thank you in advance!

**Reply by dougburks** [reply]
Looks like you've created a separate discussion for this:
https://github.com/Security-Onion-Solutions/securityonion/discussions/2028

**Comment by McSh4dow** [comment]
.
